packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+  github: GitHubAPI
+): Promise<RequiredChecksByLanguage | null> {
+  try {
+    const configFile = (
+      await github.repos.getContents({
+        owner: 'googleapis',
+        repo: 'sloth',
+        path: 'required-checks.json',
+      })
+    ).data as { content?: string };
+    return JSON.parse(
+      Buffer.from(configFile.content as string, 'base64').toString('utf8')
+    ) as RequiredChecksByLanguage;
+  } catch (err) {
+    return null;
+  }
+};
+
+mergeOnGreen.getRepoMap = async function getRepoMap(
+  github: GitHubAPI
+): Promise<Language[]> {
+  try {
+    const configFile = (
+      await github.repos.getContents({
+        owner: 'googleapis',
+        repo: 'sloth',
+        path: 'repos.json',
+      })
+    ).data as { content?: string };
+    return JSON.parse(
+      Buffer.from(configFile.content as string, 'base64').toString('utf8')
+    ).repos;
+  } catch (err) {
+    console.error(err);
+    return [];
+  }
+};
+
+mergeOnGreen.getRequiredChecks = async function getRequiredChecks(
+  github: GitHubAPI,
+  owner: string,
+  repo: string
+): Promise<string[]> {
+  const checksByLanguage = await mergeOnGreen.requiredChecksByLanguage(github);
+  const languageMap = await mergeOnGreen.getRepoMap(github);
+  if (checksByLanguage && languageMap) {
+    const language = languageMap.find(
+      (element: Language) => element.repo === `${owner}/${repo}`
+    );
+    if (language !== undefined) {
+      return checksByLanguage[language.language].requiredStatusChecks;
+    } else {
+      console.info(
+        'This repo does not have a corresponding language in sloth/repos.json'
+      );
+      return [];
+    }
+  } else {
+    console.info('Could not find any checks or a language map');
+    return [];
+  }
+};
+
+mergeOnGreen.getStatusi = async function getStatusi(
+  owner: string,
+  repo: string,
+  github: GitHubAPI,
+  headSha: string
+): Promise<CheckStatus[]> {
+  console.log('head sha ' + headSha);
+  try {
+    const data = await github.repos.listStatusesForRef({
+      owner,
+      repo,
+      ref: headSha,
+      per_page: 100,
+    });
+    return data.data;
+  } catch (err) {
+    return [];
+  }
+};
+
+mergeOnGreen.getRuns = async function getRuns(
+  owner: string,
+  repo: string,
+  github: GitHubAPI,
+  headSha: string
+): Promise<CheckRun[]> {
+  try {
+    const checkRuns = await github.checks.listForRef({
+      owner,
+      repo,
+      ref: headSha,
+      per_page: 100,
+    });
+    return checkRuns.data.check_runs;
+  } catch (err) {
+    return [];
+  }
+};
+
+mergeOnGreen.checkForRequiredSC = function checkForRequiredSC(
+  checkRuns: CheckRun[],
+  check: string
+): boolean {
+  let mergeable = false;
+  if (checkRuns.length !== 0) {
+    const checkRunCompleted = checkRuns.find(element => element.name === check);
+    if (
+      checkRunCompleted !== undefined &&
+      checkRunCompleted.conclusion === 'success'
```

Comment:
There's probably an enum available for this string

Reply:
We checked, I don't think there is unfortunately!